### DISCUSSION TITLE
The Ansible-lint compatability with Ansible is not clear from the ReadMe file and documentation

### DISCUSSION THREAD

**Discussion by elibramson** [ansible/ansible-lint#3902] [General]
We upgraded our Ansible version from **5.10.0** to  **8.5.0**. After we did it, we noticed that **Ansible-lint** (v4.3.7) doesn't work anymore since it fails with:
`NotImplementedError: an AnsibleCollectionFinder has not been installed in this process.`

It is rather obvious that we need to upgrade Ansible-lint to make it work. Still, many different environments should be upgraded, and we need help finding the needed information about Ansible and Ansible-lint compatibility. For now, we found the following facts about the Ansible-lint and Ansible compatibility:
- According to the Ansible-lint [Readme file](https://github.com/ansible/ansible-lint), Ansible-lint supports only Ansible's last two major versions.
- According to the Ansible-lint [documentation](https://ansible.readthedocs.io/projects/lint/), Ansible-lint is also supposed to help users upgrade their code to work with newer versions of Ansible. Due to this reason, it is recommended to use it with the newest version of Ansible, even if the version used in production may be older.

We can't define what that means regarding Ansible, since Ansible-lint and Ansible have different version conventions. What are Ansible's last two major versions in terms of ansible-lint? For example, let's say we will use ansible-lint v5.4.0; what ansible version does it support? 

Please tell us if there is any compliance table about this aspect or if opening an issue about it is better.

**Comment by Winand** [comment]
This does not answer your question, I just checked requirements for each ansible-lint version:
- ansible 2.9, ansible-lint [4](https://github.com/ansible/ansible-lint/blob/v4.3.7/setup.cfg)
- ansible 2.10 - 3, ansible-core 2.10, ansible-lint [5.0 - 5.1](https://github.com/ansible/ansible-lint/blob/v5.1.3/setup.cfg)
- ansible 4, ansible-core 2.11, ansible-lint [5.2 - 5.4](https://github.com/ansible/ansible-lint/blob/v5.4.0/setup.cfg)
- ansible 5, ansible-core 2.12, ansible-lint [6](https://github.com/ansible/ansible-lint/blob/v6.22.2/.config/runtime-requirements.txt)
- ansible 6 - 8, ansible-core 2.13 - 2.15, ansible-lint [24](https://github.com/ansible/ansible-lint/blob/v24.12.2/.config/requirements.in)
- ansible 9, ansible-core 2.16, ansible-lint [25.0 - 25.9](https://github.com/ansible/ansible-lint/blob/v25.9.2/pyproject.toml)
- ansible 10-13, ansible-core 2.17 - 2.20, ansible-lint [25.11](https://github.com/ansible/ansible-lint/blob/v25.11.1/pyproject.toml)
